crypto/internal/polynomial.go.tmpl

```diff
@@ -0,0 +1,25 @@
+import (
+	"github.com/consensys/gurvy/{{ toLower .Curve }}/fr"
+)
+
+// Poly polynomial represented by coefficients {{ toLower .Curve }} fr field.
+type Poly []fr.Element
+
+// Degree returns the degree of the polynomial, which is the length of Data.
+func (p Poly) Degree() uint64 {
+	res := uint64(len(p) - 1)
```

Comment:
if we have a unitialized Polynomial, (ie len(p) == 0) this return a large degree. Maybe return 0 if len(p) == 0? or panic, if we never hit that.